pmlc/dialect/tile/ir/util.cc

```diff
@@ -273,4 +274,35 @@ Value createIdentity(OpBuilder &builder, Location loc, Type elementType,
                                           getUnsignedIntegerIdentity(agg));
 }
 
+MemRefType updateMemRefWithLayoutMap(MLIRContext *context,
+                                     RankedTensorType memrefType,
+                                     Type elementType, TensorLayout layout) {
+  auto rankedTensorType = pmlc::dialect::tile::getRankedTensorType(memrefType);
+  auto originalShape = rankedTensorType.getShape();
+  auto shape = llvm::to_vector<8>(originalShape);
```

Comment:
Why do we need this?  Can't we just use rankedTensorType.getShape() or even memrefType.getShape() in the single use of 'shape' later?  Doesn't seem like rankedTensorType is really needed either, since memrefType.getRank() would also be fine.  Also is elementType just memrefType.getElementType(), in which case do we need to pass it in?  Finally, if all we are doing is changing the map on a memref type consider:

MemRefType newMemRef = MemRefType::Builder(oldMemRef).setAffineMaps({map});

Basicially MemRefType::Builder is a methods for 'modifying' the memref type.

Reply:
Will remove redundancy. Actually we only need the affine maps here and set them in the tile_to_pxa. Will change that